src/wallet/rpcwallet.cpp

```diff
@@ -3880,7 +3891,7 @@ UniValue sethdseed(const JSONRPCRequest& request)
     LOCK(pwallet->cs_wallet);
 
     // Do not do anything to non-HD wallets
-    if (!pwallet->IsHDEnabled()) {
+    if (!pwallet->IsBlank() && !pwallet->IsHDEnabled()) {
```

Comment:
In commit "[wallet] Support creating a blank wallet" (8061ea4611696b22d5d5dcdd73c1e053ffecc09d)

This is pretty convoluted. It would be more straightforward to avoid all the cancelling conditions and just write `if (!CanSupportFeature(FEATURE_HD))`

Reply:
In commit "[wallet] Support creating a blank wallet" (8061ea4611696b22d5d5dcdd73c1e053ffecc09d)

This will now allow calling `sethdseed` on a wallet with `WALLET_FLAG_DISABLE_PRIVATE_KEYS` set. That seems like a mistake because I'm not sure the resulting state would make sense, and at the very least `GenerateNewSeed` would assert false in this case.

Would suggest adding a check for WALLET_FLAG_DISABLE_PRIVATE_KEYS here.

Reply:
There's already a check for that flag above.